gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -272,6 +275,9 @@ protected void startUp() {
    * @param setStatus if true, set all jobs in the dag to pending
    */
   synchronized void addDag(Dag<JobExecutionPlan> dag, boolean persist, boolean setStatus) throws IOException {
+    if (!this.isActive) {
+      return;
+    }
```

Comment:
Would this ever cause dags to be missed? If during the process where one new dag is being added and then the leader is changed

Reply:
Oh good catch, I add the check in the load dag method to avoid this scenario.